src/TourPortal.js

```diff
@@ -178,32 +184,39 @@ class TourPortal extends Component {
       const target = document.querySelector(step.observe)
       const config = { attributes: true, childList: true, characterData: true }
       this.setState(
-        {
-          observer: new MutationObserver(mutations => {
-            mutations.forEach(mutation => {
-              if (
-                mutation.type === 'childList' &&
-                mutation.addedNodes.length > 0
-              ) {
-                const cb = () => stepCallback(mutation.addedNodes[0])
-                setTimeout(
-                  () =>
-                    this.calculateNode(
-                      mutation.addedNodes[0],
-                      step.position,
-                      cb
-                    ),
-                  100
-                )
-              } else if (
-                mutation.type === 'childList' &&
-                mutation.removedNodes.length > 0
-              ) {
-                const cb = () => stepCallback(node)
-                this.calculateNode(node, step.position, cb)
-              }
-            })
-          }),
+        prevState => {
+          if (prevState.observer) {
+            setTimeout(() => {
+              prevState.observer.disconnect()
```

Comment:
`prevState.observer.disconnect()` happens in a `setTimeout` because we need to wait for old observer's mutation callback finishes.

Reply:
Did you found any problem disconnecting without `setTimeout`?